### DISCUSSION TITLE
Unit_investment_temporal_block

### DISCUSSION THREAD

**Discussion by josefkoell** [spine-tools/SpineOpt.jl#588] [Support]
Hello everyone!

I am currently working on my model, and I was testing out the unit_investment_temporal_block feature. I have set up all the needed parameters and the model__default_investment_temporal_block, and SpineOpt is actually only activating the h2_plant (investment_unit) in the specified periods, but it is not including the unit_investment_cost. But if I use the default temporal_block, it invests in the beginning and includes the investment cost. I've included pictures of the dummy model I am trying to run below. I would be very grateful for any help!

<img width="1678" alt="Bildschirm­foto 2023-03-30 um 16 06 48" src="https://user-images.githubusercontent.com/115573136/228864753-34f963e5-dcc0-4018-b6f6-d84e60a687e9.png">
<img width="974" alt="Bildschirm­foto 2023-03-30 um 16 07 23" src="https://user-images.githubusercontent.com/115573136/228864880-91800fdb-b175-4cdc-b110-13c7ab7f8c17.png">

**Comment by DillonJ** [comment]
Hi @josefkoell 

You may need to specify initial_units_invested_available = 0. Otherwise it cheats by making the investment in the history timeslice preceding the model window

**Reply by josefkoell** [reply]
Hi @DillonJ,

Thank you for your response! I tested it, but it still does not work. It is pretty weird because if I set the unit_investment_temporal_block for the whole period, it works, but if I look at a fraction of it, it does not include the investment cost. I have built a small example. 
[Data Store 3.sqlite.zip](https://github.com/spine-tools/SpineOpt.jl/files/11113640/Data.Store.3.sqlite.zip)

**Comment by clizbe** [comment]
@josefkoell Sorry we didn't answer your question. Were you able to figure it out? (Or maybe someone can take a look again.)
